mtags/src/main/scala-2/scala/meta/internal/pc/PcDefinitionProvider.scala

```diff
@@ -131,4 +131,58 @@ class PcDefinitionProvider(val compiler: MetalsGlobal, params: OffsetParams) {
         loop(tree)
     }
   }
+
+  /** A fully attributed tree located at position `pos` */
+  private def typedTreeAt(pos: Position, unit: RichCompilationUnit): Tree = {
+    typeCheck(unit)
+    new Locator(pos).locateIn(unit.body)
```

Comment:
```suggestion
    new MetalsLocator(pos).locateIn(unit.body)
```

I think the new condition is already included there, so we should be good to use it or maybe adjust it if I am mistaken? You could drop the new Locator below.

Reply:
Yea that was my thought as well, I think we should be able to use some of the existing logic in `MetalsLocator` instead of creating a new one down below.

Reply:
Thanks!
I didn't know that we already have our own patched version of `Locator`

Reply:
Hm. I've tried to update existing `MetalsLocator` but this breaks [`CompletionInterpolationSuite.token-error`](https://github.com/scalameta/metals/blob/c2380821a14ee76fc982cfece217ae22a901288c/tests/cross/src/test/scala/tests/pc/CompletionInterpolatorSuite.scala#L467) because of having hack with `CURSOR`.

@tgodzik @ckipp01 Don't you mind to leave it as it is? It isn't clear how to properly fix it

Reply:
Which hack is that? Maybe we could put that behind a boolean and not use here?

Reply:
Upd. The problem wasn't related to `CURSOR`. The assumption that all next trees in traverser must be more narrow wasn't correct. At least might not work for `StringContext` trees as in mentioned tests-case.
I moved this logic into exiting `MetalsLocator` but changed it a little bit.